features/dd-sdk-android-profiling/src/test/kotlin/com/datadog/android/profiling/internal/PerfettoProfilerTest.kt

```diff
@@ -395,6 +396,49 @@ class PerfettoProfilerTest {
         verifyNoInteractions(mockProfilerCallback)
     }
 
+    @Test
+    fun `M not have the same stop signal W start&stop multiple times`() {
+        val inOrder = inOrder(mockService)
+
+        // When
+        // First request
+        testedProfiler.start(mockContext, setOf(fakeInstanceName))
+        val firstStopSignalCaptor = argumentCaptor<CancellationSignal>()
+        testedProfiler.stop(fakeInstanceName)
+
+        // Then
+        inOrder.verify(mockService).requestProfiling(
```

Comment:
there is a bit better API which will allow to get rid of `val inOrder = inOrder(mockService)` above:

```suggestion
        inOrder(mockService) {
          verify(mockService).requestProfiling(...)
        }
```